stream/replicator.h

```diff
@@ -163,34 +203,38 @@ class SubscribableRemoteStream final {
         return;
       }
 
-      const std::string combined_data = carried_over_data_ + chunk;
-      const auto lines = current::strings::Split<current::strings::ByLines>(combined_data);
-      size_t whole_entries_count = lines.size();
-      CURRENT_ASSERT(!combined_data.empty());
-      if (combined_data.back() != '\n' && combined_data.back() != '\r') {
-        --whole_entries_count;
-        carried_over_data_ = lines.back();
-      } else {
-        carried_over_data_.clear();
+      const size_t chunk_size = chunk.size();
+      size_t end_pos = 0;
+      if (!carried_over_data_.empty()) {
+        while (end_pos < chunk_size && chunk[end_pos] != '\n' && chunk[end_pos] != '\r') {
```

Comment:
Oh, `'\r` here and nowhere above where there are `\n` checks. Raising my eyebrows, although we could probably leave it as is for the time being as long as the tests pass.

Reply:
Where are those places, when there are `\n` check but no `\r` ones?